framework/packages/abstract-std/src/native/ibc.rs

```diff
@@ -130,6 +115,21 @@ impl IbcResult {
             PolytoneCallback::FatalError(e) => Ok(Self::FatalError(e)),
         }
     }
+
+    /// Get module to module query responses
+    pub fn get_query_result(&self, index: usize) -> StdResult<(QueryRequest<ModuleQuery>, Binary)> {
+        if let IbcResult::Query {
+            queries,
+            results: Ok(result),
+        } = &self
+        {
+            Ok((queries[index].clone(), result[index].clone()))
+        } else {
+            Err(StdError::generic_err(
+                "Failed to parse module to module query response",
+            ))
+        }
+    }
 }
```

Comment:
Do we want to verify query[index] is custom query? Because right now it's just .get(index) query response

Reply:
@CyberHoward

Reply:
Besides that LGTM

Reply:
But don't really see why we would do it for Custom but not for all the other query types?

Reply:
> We either do that for all query types or we do what we do right now and let the dev parse the result. We can add helpers to do the different query parsing later (maybe through a separate IBC query API).

Yeah, comment confused me a little

Reply:
> But don't really see why we would do it for Custom but not for all the other query types?

```
    /// Get module to module query responses
```